### PR TITLE
Fix for Wordpress cli installation

### PR DESCRIPTION
**Pull request by OZZlE**
Change the wp-cli installation to be like the recommended installation https://make.wordpress.org/cli/handbook/installing/#recommended-installation

Made wp-cli optional behind --with-wpcli flag.

Apparently the wp-cli via homebrew is deprecated: https://make.wordpress.org/cli/handbook/installing/#installing-via-homebrew 

    Error: homebrew/php was deprecated. This tap is now empty as all its formulae were migrated.

### CODE REVIEW

**Comment by FredericMartinez** [comment]
👍 💯

File: cli/Valet/DevTools.php [hunk 1]
```diff
@@ -51,17 +48,35 @@ function __construct(Brew $brew, CommandLine $cli, Filesystem $files,
...
                 $this->brew->ensureInstalled($tool, []);
             }
         }
+
+        if (!in_array('wp-cli', $skipTools)) {
+            info('Installing wp-cli...');
+            if (exec('wp --info')) {
+                info('wp-cli is already installed');
+            } else {
+                echo shell_exec(
```
Comment (abbatis) [inline cli/Valet/DevTools.php]: I wouldn't echo here, use `info()` for example

File: cli/Valet/DevTools.php [hunk 2]
```diff
@@ -51,17 +48,35 @@ function __construct(Brew $brew, CommandLine $cli, Filesystem $files,
...
             if ($this->brew->installed($tool)) {
                 info('[devtools] ' . $tool . ' already installed');
             } else {
                 $this->brew->ensureInstalled($tool, []);
             }
         }
+
+        if (!in_array('wp-cli', $skipTools)) {
+            info('Installing wp-cli...');
+            if (exec('wp --info')) {
```
Comment (abbatis) [inline cli/Valet/DevTools.php]: Use `$this->runAsUser()`?

File: cli/Valet/DevTools.php (hunk 1 above)
Comment (OZZlE) [reply cli/Valet/DevTools.php]: Alright

**Comment by OZZlE** [comment]
@abbatis fixed, how does it look now :)

**Comment by Neodork** [comment]
@OZZlE I've a few questions regarding this implementation:

- Why move away from brew? I understand that homebrew/php is deprecated however the wp-cli formulae has now moved to homebrew/core so it is in fact not deprecated on the brew side. https://formulae.brew.sh/formula/wp-cli
- If we move away from brew. Why not use the custom binary manager? https://github.com/weprovide/valet-plus/blob/master/cli/Valet/Binaries.php

**Comment by OZZlE** [comment]
@Neodork sure we can use brew if that is the case! 

However please consider that valet-plus is currently in an **non installable** state... and this fixes it in one possible way. I don't have much time to improve it further. You are free to do your own PR with this instead of course though.

But code can always be improved of course. This could be one way to make valet-plus at least work and not be broken.

Also it's strange that Wordpress'es own documentation is outdated..

**Comment by Neodork** [comment]
@OZZlE Valid points raised and I agree on the installable state being a big issue. Tell you what I will look into the WP-CLI installation today and if necessary move from brew to Binaries. However I prefer to stay with brew because the brew implementation automatically updates the version which makes it less of a bother to keep up-to-date.

**Comment by Neodork** [comment]
@OZZlE Please see my reaction in https://github.com/weprovide/valet-plus/issues/179#issuecomment-397351200 can you confirm the install is now working?

**Comment by samgranger** [comment]
I assume this PR can get closed then @OZZlE ?

**Comment by samgranger** [comment]
Closing now, please create a ticket in case I'm mistaken!

**Pull request closed without merging**